bwt/wavelet.go

```diff
@@ -0,0 +1,356 @@
+package bwt
+
+import (
+	"fmt"
+	"math"
+
+	"golang.org/x/exp/slices"
+)
+
+/*
+
+For the waveletTree's usage, please read the its
+method documentation. To understand what it is and how
+it works for either curiosity or maintenance, then read below.
+
+# WaveletTree
+
+The Wavelet Tree allows us to conduct RSA queries on strings. in
+a memory and run time efficient manner.
+RSA stands for (R)ank, (S)elect, (A)ccess.
+
+## The Character's Path Encoding
+
+One important component is a character's path encoding.
+Which character we are working with in a given path in the tree.
+For example, given the alphabet A B C D E F G, a possible encoding is:
+
+A: 000
+B: 001
+C: 010
+D: 011
+E: 100
+F: 101
+G: 110
+H: 111
+
+If we wanted to get to the leaf that represent the character D, we'd
+take the path:
+
+   root
+  /
+left
+  \
+ right
+    \
+   right
+
+## The Data Represented at each node
+
+Let us consider the sequence "bananas"
+It has the alphabet b, a, n, s
+Let's say it has the encoding:
+a: 00
+n: 01
+b: 10
+s: 11
+and that 0 if left and 1 is right
+We can represent this tree with bitvectors:
+
+     0010101
```

Comment:
```suggestion
     0010101
     bananas
```
I think including the string translation of the bitvector as Alex does in [this blog post](https://www.alexbowe.com/wavelet-trees/) would be very helpful!